### PR TITLE
Implementation of user session

### PR DESCRIPTION
**Pull request by AreYouIvan**
Hello, 
The implementation of the user session is finished. 
You will see a lot of changes in these 20 files some of them will be:

- The authentication and session logic, to make this works I have used ReactContext, ReactHooks, Render conditionals, and LocalStorage from JavaScript.
- I modified some files to add buttons, and little functionalities
- Also, some changes are just in the format, e.g.: Adding just a semicolon at the end of the line.

This PR solves the #67

image

image

image

### CODE REVIEW

File: src/components/UserSessionValidation/index.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import React, { useContext } from "react";
+import { UserContext } from "../../context/UserContext";
+import { Navigate } from "react-router-dom";
+
+
+const UserSessionValidation = () => {
+  // const { login, userSession } = useContext(UserContext)
+  
+  return (
+    <>{!window.localStorage.getItem("userSession") && <Navigate to="/" replace={true} />}</>
```
Comment (LilianaRestrepoTorres) [inline src/components/UserSessionValidation/index.js:10]: :rocket:

File: src/pages/Home/index.js [hunk 2]
```diff
@@ -2,25 +2,40 @@ import React, { useState, useEffect, useContext } from "react";
...
-  useEffect(async () => {
-    const response = await axios.get("https://ghibliapi.herokuapp.com/films");
-    setFilms(response.data);
+
+  useEffect(() => {
+    let isSubscribed = true;
+    axios
+      .get("https://ghibliapi.herokuapp.com/films")
+      .then(response => {
+        if (isSubscribed) {
```
Comment (LilianaRestrepoTorres) [inline src/pages/Home/index.js:22]: Is this validation to make sure that only is executed once?

**Review by LilianaRestrepoTorres — APPROVED** [review]

File: src/components/Film/index.js [hunk 3]
```diff
@@ -16,42 +16,53 @@ function Film() {
   // console.log("film in film page", film[0].title);
   if (film) {
     return (
-      <div className="film-component">
-        <div className="film-container">
-          <div className="film-head">
-            <div className="film-title">
-              <h1>{film.title}</h1>
-              <h2>{film.release_date}</h2>
+      <>
```
[the remark this answers is not in the document]
Comment (AreYouIvan) [reply src/components/Film/index.js:19]: It was a mistake, before the built of the userSessionValidation I was trying to make the validation with a conditional written in every component, to make this I was need to execute the function before the render of the component, that's the reason. Is not necessary already so, I'm going to delete it. Thank you.

File: src/components/FacebookLogin/index.js [hunk 4]
```diff
@@ -1,29 +1,39 @@
...
+        // setLoginError(error.response.data.message)
+        // setTimeout(() => {
+        //   setLoginError(false)
+        // }, 2000)
+      })
+  }
   return (
     <div>
-      <FacebookLogin
-        appId={process.env.REACT_APP_FACEBOOK_API_KEY}
```
[the remark this answers is not in the document]
Comment (AreYouIvan) [reply src/components/FacebookLogin/index.js:16]: Got it, in a moment.

File: src/pages/Home/index.js (hunk 2 above)
Comment (AreYouIvan) [reply src/pages/Home/index.js:22]: Of course. Besides we're already receiving errors and handle them, this keep being async and is also executed once the component is rendered due the useEffect. 

That's the difference from the other.

**Comment by AreYouIvan** [comment]
@LilianaRestrepoTorres 
.env was deleted and the React fragment removed, the Facebook login is pending. :)

File: src/pages/Home/index.js [hunk 5]
```diff
@@ -2,25 +2,40 @@ import React, { useState, useEffect, useContext } from "react";
...
   // fetch Data
   const [films, setFilms] = useState([]);
-  useEffect(async () => {
-    const response = await axios.get("https://ghibliapi.herokuapp.com/films");
-    setFilms(response.data);
+
+  useEffect(() => {
+    let isSubscribed = true;
+    axios
+      .get("https://ghibliapi.herokuapp.com/films")
```
Comment (Hai5edfm) [inline src/pages/Home/index.js:20]: :v duda dudosa, donde es que la API te solicita el access_token para hacer una request? :v, es decir, si, a esto el usuario tiene acceso, pero como sabes que no es un usuario baneado xd o algo asi, con el isSuscribed?, en ese caso esta harcodeado?

Comment (Hai5edfm) [reply src/pages/Home/index.js:20]: o el navegador ya sabe acceder automaticamente al token? :v

**Review by Hai5edfm — APPROVED** [review]

File: src/pages/Home/index.js (hunk 5 above)
Comment (AreYouIvan) [reply src/pages/Home/index.js:20]: Yes as @LilianaRestrepoTorres said. That piece of code is fetching only the movies and is not our API as well.
The variable isSubscribed was an attempt to fix a warning in home page.

**Pull request merged**